kotlin-ktor/src/test/java/com/baeldung/graphql/client/GraphQLClientTest.kt

```diff
@@ -0,0 +1,142 @@
+        AttendeeRepository.clear()
+    }
+
+    private suspend fun createDummyConference(attendees: List<Int> = listOf()) {
+        graphQLService.createOrSaveConference(null, "Some Conference", attendees)
+    }
+
+    private suspend fun createDummyAttendee() {
+        graphQLService.createOrSaveAttendee(null, "Some Attendee")
+    }
+
+    @Test
+    fun `given conference by id when fetching existing conference then should fetch it`(): Unit = runBlocking {
+        createDummyConference()
+        val fetchedConference = graphQLService.findConferenceById(0)
+        assertEquals(0, fetchedConference?.id)
+        assertEquals("Some Conference", fetchedConference?.name)
+    }
+
+    @Test
+    fun `given conference by id when fetching not existing conference then should return null`(): Unit = runBlocking {
+        val fetchedConference = graphQLService.findConferenceById(0)
+        assertNull(fetchedConference)
+    }
+
+    @Test
+    fun `given create or save conference when saving not existing conference then should create it`(): Unit = runBlocking {
+        graphQLService.createOrSaveConference(null, "Some Conference")
+        val createdConference = ConferenceRepository.findById(0)
+        assertEquals("Some Conference", createdConference?.name)
+    }
+
+    @Test
+    fun `given create or save conference when saving existing conference then should update it`(): Unit = runBlocking {
+        createDummyConference()
+        graphQLService.createOrSaveConference(0, "Same Conference")
+        val updatedConference = ConferenceRepository.findById(0)
+        assertEquals("Same Conference", updatedConference?.name)
+    }
+
+    @Test
+    fun `given create or save attendee when saving not existing attendee then should create it`(): Unit = runBlocking {
+        graphQLService.createOrSaveAttendee(null, "Some Attendee")
+        val createdAttendee = AttendeeRepository.findById(0)
+        assertEquals("Some Attendee", createdAttendee?.name)
+    }
+
+    @Test
+    fun `given create or save attendee when saving existing attendee then should update it`(): Unit = runBlocking {
+        createDummyAttendee()
+        graphQLService.createOrSaveAttendee(0, "Same Attendee")
+        val updatedAttendee = AttendeeRepository.findById(0)
+        assertEquals("Same Attendee", updatedAttendee?.name)
+    }
+
+    @Test
+    fun `given get object by id when fetching conference then should return it`(): Unit = runBlocking {
+        createDummyConference()
+        val objectById = graphQLService.getObjectById(0)
+        assertIs<com.baeldung.graphql.client.generated.objectbyidquery.Conference>(objectById)
+    }
+
+    @Test
+    fun `given get object by id when fetching attendee then should return it`(): Unit = runBlocking {
+        createDummyAttendee()
+        val objectById = graphQLService.getObjectById(0)
+        assertIs<com.baeldung.graphql.client.generated.objectbyidquery.Attendee2>(objectById)
```

Comment:
Interesting, this `Attendee2` highlights even more this issue of generating many types with the same name that vary slightly depending on the area they were generated from. I'm surprised this issue/limitation hasn't received more attention so far. No action here, just commenting.